presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -323,9 +323,8 @@ private boolean isTablesEnumeratingTable(SchemaTableName schemaTableName)
         }
 
         Session session = ((FullConnectorSession) connectorSession).getSession();
-        return metadata.listSchemaNames(session, catalogName).stream()
-                .filter(schema -> !predicate.isPresent() || predicate.get().test(schemaAsFixedValues(schema)))
-                .map(schema -> new QualifiedTablePrefix(catalogName, schema))
+        return listSchemaNames(session).stream()
+                .filter(prefix -> !predicate.isPresent() || predicate.get().test(schemaAsFixedValues(prefix.getSchemaName().get())))
```

Comment:
predicate can take catalog name, here you're providing only schema name even though catalog name is known

(this is just a thought, only vaguely related to `Extract InformationSchemaMetadata#listSchemaNames` commit)

Reply:
> predicate can take catalog name, here you're providing only schema name even though catalog name is known

Information schema is catalog specific. Does it make sense to write queries like:
```
SELECT .. from catalog_foo.information_schema.tables WHERE table_catalog = 'catalog_bar';
```
?

Reply:
To me -- it doesn't, but still may happen.
-- Also, there is `system.jdbc.tables`.